### ISSUE TITLE
FbaInventoryApi::getInventorySummaries InvalidInput error 

### ISSUE THREAD

**Issue by tawfekov** [jlevers/selling-partner-api#75]
Hello 
I just faced a new issue when using this, its not very clear error  : 
```
 SellingPartnerApi\Api\FbaInventoryApi::getInventorySummaries("Marketplace", "ATVPDKIKX0DER", "2021-08-16T04:00:00Z")
```

I'm getting this error 
```
 SellingPartnerApi\ApiException

  [400] {
  "errors": [
    {
      "code": "InvalidInput",
      "message": "Invalid Input",
      "details": ""
    }
  ]
}
```

I'm using `3.1.2` now , any help ? 

Thanks

**Comment by tawfekov** [comment]
setting the param `$details` to `false` will make the request successful 
its default value is `false` , you will face this error only when you are looking for inventory summary , 
for my case : I'm looking for it :(

**Comment by jlevers** [comment]
Can I see your code? I think you might have the parameter order a little bit wrong, based on what you wrote in your OP.

**Comment by tawfekov** [comment]
my code is very simple 

```
        $config = getSpApiConfig();
        // get inventory api instance
        $fbaInventoryApi = new FbaInventoryApi($config);
        // api request params
        $startDateTime = $all ? null : Carbon::today()->startOfDay()->toIso8601ZuluString();
        return $fbaInventoryApi->getInventorySummaries(
            'Marketplace',
            config('sp-api.granularity_id'),
            config('sp-api.marketplace_ids'),
            false,
            $startDateTime,
            null,
            $next_token
        );
```

**Comment by jlevers** [comment]
This code is working for me:

```php
use SellingPartnerApi\Api\FbaInventoryApi;

$config = _loadConfig();  // Just to shorten things...

$api = new FbaInventoryApi($config);
$res = $api->getInventorySummaries('Marketplace', ['ATVPDKIKX0DER'], ['ATVPDKIKX0DER']);
```

Are you actually passing a value in `$next_token`?

**Comment by tawfekov** [comment]
pardon me , maybe I had to mention in my code the `details` params  to be `true` , it will break

```php
        $config = getSpApiConfig();
        // get inventory api instance
        $fbaInventoryApi = new FbaInventoryApi($config);
        // api request params
        $startDateTime = $all ? null : Carbon::today()->startOfDay()->toIso8601ZuluString();
        return $fbaInventoryApi->getInventorySummaries(
            'Marketplace',
            config('sp-api.granularity_id'),
            config('sp-api.marketplace_ids'),
            true,
            $startDateTime,
            null,
            $next_token
        );
```

however I will test it once I arrive and update you back 

Thanks
Tawfek

**Comment by jlevers** [comment]
Ah, I see -- yes, I got `InvalidInput` when I set `$details` to true. I recommend opening an issue in [amzn/selling-partner-api-docs](https://github.com/amzn/selling-partner-api-docs), because I think this is a problem on their end.

**Comment by tawfekov** [comment]
Thank you for your time @jlevers !

**Comment by allen1706** [comment]
Parameter $details should be passed in as a String instead of Boolean.

**Comment by jlevers** [comment]
Thanks @allen1706, good catch! This is still a bug with Amazon's OpenAPI models, but I was able to modify the library to accomodate their error. The fix is deployed in `v4.1.6`.
